ui/app/AppLayouts/Communities/panels/MintTokensSettingsPanel.qml

```diff
@@ -257,381 +112,187 @@ SettingsPageLayout {
                 }
             }
-
-            Binding {
-                target: root
-                property: "title"
-                value: optionsTab.currentItem == collectiblesTab
-                       ? d.newCollectiblePageTitle : d.newAssetPageTitle
-                restoreMode: Binding.RestoreBindingOrValue
-            }
         }
     }
 
     Component {
         id: previewTokenView
 
-        CommunityTokenView {
-            id: preview
-
-            function signMintTransaction() {
-                root.setFeeLoading()
-                if(preview.isAssetView)
-                    root.mintAsset(token)
-                else
-                    root.mintCollectible(token)
-
-                root.resetNavigation()
-            }
-
-            viewWidth: root.viewWidth
-
-            onMintClicked: signMintPopup.open()
-
-            Binding {
-                target: root
-                property: "title"
-                value: preview.name
-            }
+        SettingsPage {
+            id: tokenPreviewPage
 
-            Binding {
-                target: root
-                property: "subTitle"
-                value: preview.symbol
-                restoreMode: Binding.RestoreBindingOrValue
-            }
+            property alias token: preview.token
 
-            SignTokenTransactionsPopup {
-                id: signMintPopup
+            pageTitle: token.name
+            pageSubtitle: token.symbol
 
-                anchors.centerIn: Overlay.overlay
-                title: qsTr("Sign transaction - Mint %1 token").arg(signMintPopup.tokenName)
-                tokenName: preview.name
-                accountName: preview.accountName
-                networkName: preview.chainName
-                feeText: root.feeText
-                errorText: root.errorText
-                isFeeLoading: root.isFeeLoading
+            contentItem: CommunityTokenView {
+                id: preview
 
-                onOpened: {
+                function signMintTransaction() {
                     root.setFeeLoading()
-                    root.signMintTransactionOpened(preview.chainId, preview.accountAddress)
-                }
-                onCancelClicked: close()
-                onSignTransactionClicked: preview.signMintTransaction()
-            }
-        }
-    }
-
-    Component {
-        id: mintTokenFooter
 
-        MintTokensFooterPanel {
-            id: footerPanel
-
-            readonly property bool deployStateCompleted: !!d.currentToken
-                                                         ? d.currentToken.deployState === Constants.ContractTransactionStatus.Completed
-                                                         : false
-
-            function closePopups() {
-                remotelyDestructPopup.close()
-                alertPopup.close()
-                signTransactionPopup.close()
-                burnTokensPopup.close()
-            }
+                    if(preview.isAssetView)
```

Comment:
Here maybe we can avoid this difference and just have a `mint(token)` signal, since token object contains information about if it's an asset or a collectible indeed.